test/unity/toolkits/object_detection/test_od_data_iterator.cxx

```diff
@@ -119,6 +123,146 @@ BOOST_AUTO_TEST_CASE(test_simple_data_iterator) {
   assert_batch(batch, 0);
 }
 
+//=============================================================================
+
+BOOST_AUTO_TEST_CASE(test_simple_data_iterator_coordinates) {
+
+
+  auto create_data_opts = [](const std::string image_origin,
+                         const std::string annotation_scale,
+                         const std::string annotation_origin,
+                         float x, float y, float w, float h) {  
+
+    data_iterator::parameters result;
+
+    flex_list images(1);
+    flex_list annotations(1);
+    std::vector<unsigned char> buffer(IMAGE_HEIGHT * IMAGE_WIDTH * 3);
+
+    // Each pixel has R, G, and B value equal to the row index (modulo 256).
+    std::fill(buffer.begin(), buffer.end(),
+              static_cast<unsigned char>(0 % 256));
+    images[0] = flex_image(reinterpret_cast<char*>(buffer.data()), IMAGE_HEIGHT,
+                           IMAGE_WIDTH, 3, buffer.size(),
+                           IMAGE_TYPE_CURRENT_VERSION,
+                           static_cast<int>(Format::RAW_ARRAY));
+
+     
+
+    if (image_origin == "top_left") {
+        std::cout << "image : top_left" <<std::endl;
+        result.image_origin = data_iterator::image_origin_enum::TOP_LEFT;
+    }
+    else if (image_origin == "bottom_left") {
+        std::cout << "image : bottom_left" <<std::endl;
+        result.image_origin = data_iterator::image_origin_enum::BOTTOM_LEFT;
+    }
+    else {
+        // Add error logging
```

Comment:
Do you want to do that as part of this PR? Or should this be a todo? Same question for the other two similar spots in this function.

Reply:
Same as above